base/linalg/rowvector.jl

```diff
@@ -147,6 +154,11 @@ end
 
 # Multiplication #
 
+# inner product -> dot product specializations
+@inline *{T<:Real}(rowvec::RowVector{T}, vec::AbstractVector{T}) = dot(parent(rowvec), vec)
+@inline *(rowvec::ConjRowVector, vec::AbstractVector) = dot(rowvec', vec)
```

Comment:
The first is true, but not the way you think: you end up on [this line](https://github.com/andyferris/julia/blob/9e0e1cf6641aa0c609c4efa3f4e8f9ccb0dbd536/base/linalg/rowvector.jl#L168) so you get `*(rowvec, matrix) = RowVector(At_mul_B(matrix, parent(rowvec)))`. Would that be a performance penalty?

I should chase down all the conjugated cases and make sure they do something sensible. I think what happens is a conjugated copy of the parent vector is created as a temporary, and the above gets called. (We should be able to remove this temporary).